### PR TITLE
Tweak the initial test run process factory

### PR DESCRIPTION
**Pull request by theofidry**
- Rename `InitialTestRunBuilder` to `InitialTestsRunFactory` (with the `s` to be consistent with the runner `InitialTestsRunner`)
- Remove the optional parameters from `InitialTestRunBuilder` & `InitialTestsRunner`: they do not need to be optional
- Move the factories from `Infection\Process\Builder` to `Infection\Process\Factory`
- Consolidate the tests. Notable change: I no longer mock `Process` in `InitialTestsRunner` but instead favoured an integration test there

### CODE REVIEW

File: src/Engine.php [hunk 1]
```diff
@@ -130,8 +130,8 @@ private function runInitialTestSuite(): void
 
         $initialTestSuitProcess = $this->initialTestsRunner->run(
             $this->config->getTestFrameworkExtraOptions(),
-            $this->config->shouldSkipCoverage(),
-            explode(' ', (string) $this->config->getInitialTestsPhpOptions())
+            explode(' ', (string) $this->config->getInitialTestsPhpOptions()),
```
Comment (maks-rafalko) [inline src/Engine.php]: I'm not sure this change is needed (swapping parameters)

Comment (theofidry) [reply src/Engine.php]: no but I also dislike having the parameters declared in a certain order in the method and another order for the properties declaration, assignment and method orders, so just made sure they are all in the same order

Comment (theofidry) [reply src/Engine.php]: Actually that's not the case here so nvm, will revert that

Comment (theofidry) [reply src/Engine.php]: It was to keep the same order as `TestFrameworkAdapter::getInitialTestRunCommandLine`

**Pull request merged**